examples/layout_padded/layout_padded.cpp

```diff
@@ -0,0 +1,648 @@
+// This makes it possible, for example, to use aligned SIMD instructions
+// without needing a loop prelude or postlude.
+//
+// If all the extents are stored as compile-time ("static") values,
+// the strides will be computable at compile time.
+//
+// Preconditions:
+//
+// overalignment_factor must be either 1, or a (positive integer) power of two.
+template<StorageOrder storage_order, std::size_t overalignment_factor>
+struct layout_padded {
+  static_assert(details::ceiling_power_of_two(overalignment_factor) == overalignment_factor,
+		"overalignment_factor must be a power of 2");
+
+  template <class Extents>
+  class mapping {
+  private:
+    using inner_layout_type = details::storage_order_to_layout_t<storage_order>;
+    static constexpr std::integral_constant<std::size_t, overalignment_factor> overalignment{};
+    static constexpr std::integral_constant<StorageOrder, storage_order> order{};
+    using padded_extents_type =
+      decltype(pad_extents(std::declval<Extents>(), order, overalignment));
+    using inner_mapping_type = typename inner_layout_type::template mapping<padded_extents_type>;
+
+  public:
+    using extents_type = Extents;
+    using index_type = typename extents_type::index_type;
+    using size_type = typename extents_type::size_type;
+    using rank_type = typename extents_type::rank_type;
+    using layout_type = layout_padded<storage_order, overalignment_factor>;
+
+    // layout_stride::mapping deliberately only defines the copy constructor
+    // and copy assignment operator.
+    // This is fine, because all the storage is std::array-like;
+    // there's no advantage to move construction or move assignment.
+    // We imitate this.
+    MDSPAN_INLINE_FUNCTION_DEFAULTED
+    constexpr mapping() noexcept = default;
+    MDSPAN_INLINE_FUNCTION_DEFAULTED
+    constexpr mapping(mapping const&) noexcept = default;
+    MDSPAN_INLINE_FUNCTION_DEFAULTED _MDSPAN_CONSTEXPR_14_DEFAULTED
+    mapping& operator=(const mapping&) noexcept = default;
+
+    MDSPAN_INLINE_FUNCTION
+    constexpr mapping(const extents_type& e) :
+      extents_(e), inner_mapping_(pad_extents(e, order, overalignment))
+    {}
+
+    MDSPAN_INLINE_FUNCTION
+    constexpr const extents_type& extents() const noexcept
+    {
+      return extents_;
+    }
+
+    MDSPAN_INLINE_FUNCTION
+    constexpr std::array<index_type, extents_type::rank()>
+    strides() const noexcept
+    {
+      return inner_mapping_.strides();
+    }
+
+    MDSPAN_INLINE_FUNCTION
+    constexpr index_type required_span_size() const noexcept
+    {
+      return inner_mapping_.required_span_size();
+    }
+
+    MDSPAN_TEMPLATE_REQUIRES(
+      class... Indices,
+      /* requires */ (sizeof...(Indices) == Extents::rank() &&
+        _MDSPAN_FOLD_AND(_MDSPAN_TRAIT(std::is_convertible, Indices, index_type) /*&& ...*/ ) &&
+	_MDSPAN_FOLD_AND(_MDSPAN_TRAIT(std::is_nothrow_constructible, index_type, Indices) /*&& ...*/)
+      )
+    )
+    MDSPAN_FORCE_INLINE_FUNCTION
```

Comment:
Why force inline here?

Reply:
@dalg24 It's what the other layouts do : - )

Reply:
@dalg24 I changed all uses of `MDSPAN_FORCE_INLINE_FUNCTION` in this PR to `MDSPAN_INLINE_FUNCTION`.